virttest/qemu_vm.py

```diff
@@ -2222,22 +2222,33 @@ def add_secure_guest_descriptor(params):
             else:
                 # TODO: Use QDevices for this and set the addresses properly
                 for sound_device in soundhw.split(","):
+                    devs = []
                     if "hda" in sound_device:
-                        intel_hda = QDevice("intel-hda", parent_bus=parent_bus)
-                        set_cmdline_format_by_cfg(
-                            intel_hda, self._get_cmdline_format_cfg(), "soundcards"
-                        )
-                        devices.insert(intel_hda)
-                        dev = QDevice("hda-duplex")
+                        intel_hda = QDevice('intel-hda', parent_bus=parent_bus)
+                        devs.append(intel_hda)
+                        # FIXME, dummy audiodev backend driver at current stage, and hardcode the id to audiodev1
+                        if Flags.AUDIODEV in devices.caps:
+                            audiodev_id = "audiodev1"
+                            audiodev = StrDev('audiodev', cmdline="-audiodev '{\"id\":\"%s\",\"driver\":\"none\"}'" % audiodev_id)
```

Comment:
At first i introduce an `QAudioDev` in `qdevices` to support  backends list and the properties for different backend  and the `_cmdline_json`.   
However, back to the `qemu_vm` to add `soundhw`:   `  if "hda" in sound_device: `, the `hda-duplex` always be added, no matter what sound device be specified with `soundcards = ****`, such as hda-micro, hda-output.
Which means the backend variable we specified for different soundcard never be used.
The background for this design (use hda-duplex all the time) is unclear for me,  i do not want to break it now.
And we only need the dummy audiodev backend now.  So i remove the  over-engineered code and  hardcode the json command line for `audiodev` finally.

@luckyh Do you have any comments for this ?